keps/sig-auth/3304-client-go-tls-config/README.md

```diff
@@ -0,0 +1,607 @@
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests for meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+`k8s.io/client-go` serves as the foundation for Kubernetes client side tooling written in Go.  While
+it is often used to communicate with the Kubernetes API server, it is also used to interact with
+other servers such as admission webhooks.  Server components like the API server provide server side
+configuration of the minimum valid TLS version and the set of TLS cipher suites that may be used.
+This KEP proposes the client side equivalent of that configuration.
+
+## Motivation
+
+Clients written using `k8s.io/client-go` should be able to assert control over their TLS configuration
+just as servers are able to.  Today `k8s.io/client-go` provides no control over the cipher suites used
+and the TLS minimum version is hard coded to TLS 1.2.
+
+### Goals
+
+- When TLS 1.2 is used, control over the set of cipher suites that may be used
+- The ability to set TLS 1.3 as the minimum version that will be used
+- The process wide default should be configurable globally
+- Individual clients should be able to override that process default
+
+### Non-Goals
+
+- Supporting TLS versions less than 1.2
+- Changing the default TLS configuration of `k8s.io/client-go`
+- Allowing the use of insecure ciphers with TLS 1.2
+- Changes to any user facing API such as the kubeconfig format, kubectl CLI flags, etc.
```

Comment:
Can you provide a reason for making this a non-goal?  It's not immediately obvious to me that this should be code-level only.

Reply:
Wouldn't it be cool if we added the TLS configuration flags to the kubectl CLI? I think this could be a low hanging fruit and would make this proposal more "complete".